verifier/providerverifier.go

```diff
@@ -0,0 +1,280 @@
+package verifier
+
+import (
+	"context"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/openpubkey/openpubkey/gq"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+	oidcclient "github.com/zitadel/oidc/v2/pkg/client"
+)
+
+type ProviderVerifier struct {
+	issuer          string
+	commitmentClaim string
+	options         ProviderVerifierOpts
+}
+
+type ProviderVerifierOpts struct {
+	// If ClientID is specified, then verification will require that the ClientID
+	// be present in the audience ("aud") claim of the PK token payload
+	ClientID string
+	// Custom function for discovering public key of Provider
+	DiscoverPublicKey func(ctx context.Context, kid string, issuer string) (jwk.Key, error)
+	// Allows for successful verification of expired tokens
+	SkipExpirationCheck bool
+}
+
+// Creates a new ProviderVerifier with required fields
+//
+// issuer: Is the OpenID provider issuer as seen in ID token e.g. "https://accounts.google.com"
+// commitmentClaim: the ID token payload claim name where the cicHash was stored during issuance
+func NewProviderVerifier(issuer, commitmentClaim string, options ProviderVerifierOpts) *ProviderVerifier {
```

Comment:
We should provide pre-configured ProviderVerifier for at least Google and Github, similar to how we provide client.providers.Google. This would let us hide the commitmentClaim.